core/src/main/java/com/tweet/core/model/User.java

```diff
@@ -13,6 +12,10 @@
     private String firstName;
     private String surname;
 
+    @Enumerated(EnumType.STRING)
+    @ManyToMany
```

Comment:
This may not be enough, I would suggest something like this:
@ManyToMany(cascade = CascadeType.ALL)
@JoinTable(name = "user_role", joinColumns = @JoinColumn(name = "user_id"), inverseJoinColumns = @JoinColumn(name = "role_id"))

Reply:
This kind of table join actually creates additional table, it will look like this (more less)

CREATE TABLE USER_ROLE (
  role_id INT unsigned NOT NULL AUTO_INCREMENT,
  role VARCHAR2(32) NOT NULL
);

CREATE TABLE ROLE_MAPPING (
  user_id INT NOT NULL,
  role_id INT NOT NULL,
  PRIMARY KEY(user_id, role_id)
);

Reply:
Actually it may be good idea to add to UserRole annotation @Table(name = "ROLE")
This was we could have more readable schema:
CREATE TABLE ROLE (
  role_id INT unsigned NOT NULL AUTO_INCREMENT,
  role VARCHAR2(32) NOT NULL
);

CREATE TABLE USER_ROLE (
  user_id INT NOT NULL,
  role_id INT NOT NULL,
  PRIMARY KEY(user_id, role_id)
);

and out joining annotations will look like this:
    @ManyToMany(cascade = CascadeType.ALL)
    @JoinTable(name = "user_role", joinColumns = @JoinColumn(name = "user_id"), inverseJoinColumns = @JoinColumn(name = "role_id"))